Sources/MuxStatsGoogleIMAPlugin/MuxImaListener.m

```diff
@@ -64,58 +64,100 @@ - (void)setupAdViewData:(MUXSDKAdEvent *)event withAd:(IMAAd *)ad {
+
+        // TODO: use newer IMA API here. universalAdIdValue
+        // is deprecated, but used for time being for parity
+        // with web&android
+        adData.adUniversalId = event.ad.universalAdIdValue;
+    }
+
+    return [self dispatchEvent:event.type
+                    withAdData:adData
+                 withIMAAdData:event.adData];
+}
+
+- (nullable MUXSDKAdEvent *)dispatchEvent:(IMAAdEventType)eventType
+                               withAdData:(nullable MUXSDKAdData *)adData
+                            withIMAAdData:(nullable NSDictionary *)imaAdData {
     MUXSDKAdEvent *playbackEvent;
-    
-    NSDictionary *adData = event.adData;
-    
-    switch(event.type) {
-        case kIMAAdEvent_STARTED:
+
+    switch(eventType) {
+        case kIMAAdEvent_STARTED: {
             if (_sendAdplayOnStarted) {
-                playbackEvent = [MUXSDKAdPlayEvent new];
+                playbackEvent = [[MUXSDKAdPlayEvent alloc] init];
                 [_playerBinding dispatchAdEvent: playbackEvent];
             } else {
                 _sendAdplayOnStarted = YES;
             }
-            playbackEvent = [MUXSDKAdPlayingEvent new];
+            playbackEvent = [[MUXSDKAdPlayingEvent alloc] init];
             break;
-        case kIMAAdEvent_FIRST_QUARTILE:
-            playbackEvent = [MUXSDKAdFirstQuartileEvent new];
+        }
+        case kIMAAdEvent_FIRST_QUARTILE: {
+            playbackEvent = [[MUXSDKAdFirstQuartileEvent alloc] init];
+            break;
+        }
+        case kIMAAdEvent_MIDPOINT: {
+            playbackEvent = [[MUXSDKAdMidpointEvent alloc] init];
             break;
-        case kIMAAdEvent_MIDPOINT:
-            playbackEvent = [MUXSDKAdMidpointEvent new];
+        }
+        case kIMAAdEvent_THIRD_QUARTILE: {
+            playbackEvent = [[MUXSDKAdThirdQuartileEvent alloc] init];
             break;
-        case kIMAAdEvent_THIRD_QUARTILE:
-            playbackEvent = [MUXSDKAdThirdQuartileEvent new];
+        }
+        case kIMAAdEvent_SKIPPED: {
+            playbackEvent = [[MUXSDKAdEndedEvent alloc] init];
             break;
-        case kIMAAdEvent_SKIPPED:
-        case kIMAAdEvent_COMPLETE:
-            playbackEvent = [MUXSDKAdEndedEvent new];
+        }
+        case kIMAAdEvent_COMPLETE: {
+            playbackEvent = [[MUXSDKAdEndedEvent alloc] init];
             break;
-        case kIMAAdEvent_PAUSE:
-            playbackEvent = [MUXSDKAdPauseEvent new];
+        }
+        case kIMAAdEvent_PAUSE: {
+            playbackEvent = [[MUXSDKAdPauseEvent alloc] init];
             break;
-        case kIMAAdEvent_RESUME:
-            playbackEvent = [MUXSDKAdPlayEvent new];
-            [self setupAdViewData:playbackEvent withAd:event.ad];
+        }
+        case kIMAAdEvent_RESUME: {
+            playbackEvent = [[MUXSDKAdPlayEvent alloc] init];
+            MUXSDKViewData *viewData = [[MUXSDKViewData alloc] init];
+            if ([_playerBinding getCurrentPlayheadTimeMs] < 1000) {
```

Comment:
question: I remember customers asking for this value to be configurable. Does anyone recall where we landed on that? (@andrewjl-mux @daytime-em)

Reply:
Makes sense to do. Better as a separate change.